File: app/src/main/java/com/qns/data/local/entity/MessageEntity.java
package com.qns.data.local.entity;
import androidx.annotation.NonNull;
import androidx.room.*;
@Entity(tableName = "messages",
    indices = { @Index("chatId"), @Index({"chatId","createdAt"}) })
public class MessageEntity {
    @PrimaryKey @NonNull public String id = "";
    public String chatId, senderId, clientMessageId, encryptedContent, ratchetHeader, decryptedCache, decryptionError;
    public int protocolVersion = 1;
    public boolean decryptionFailed = false, delivered = false, read = false, isMine = false;
    public long createdAt;
}

File: qunes-android-app/src/main/kotlin/com/qunes/app/data/local/entities/MessageEntity.kt
package com.qunes.app.data.local.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "messages")
data class MessageEntity(
    @PrimaryKey val id: String,
    val chatId: String,
    val senderId: String,
    val payload: String, // Зашифровано Kyber
    val iv: String,
    val signature: String,
    val type: String,
    val timestamp: Long,
    val isOutgoing: Boolean
)